src/psyclone/psyir/transformations/block_loop_trans.py

```diff
@@ -0,0 +1,237 @@
+        :raises TransformationError: if the supplied Loop has is a blocked
+                loop.
+        :raises TransformationError: if the supplied Loop has a step size \
+                of 0.
+        :raises TransformationError: if the supplied Loop writes to Loop \
+                variables inside the Loop body.
+        '''
+        super(BlockLoopTrans, self).validate(node, options=options)
+        # If step is a variable don't support it for now.
+        if options is None:
+            options = {}
+        if isinstance(node.children[2], nodes.Reference):
+            raise TransformationError("Cannot apply a BlockLoopTrans to "
+                                      "a loop with a non-constant step size")
+        block_size = options.get("blocksize", 32)
+        if abs(int(node.children[2].value)) > block_size:
+            raise TransformationError("Cannot apply a BlockLoopTrans to "
+                                      "a loop with larger step size than the "
+                                      "chosen block size")
+        if 'blocked' in node.annotations:
+            raise TransformationError("Cannot apply a BlockLoopTrans to "
+                                      "an already blocked loop.")
+
+        if int(node.children[2].value) == 0:
+            raise TransformationError("Cannot apply a BlockLoopTrans to "
+                                      "a loop with a step size of 0")
+        # Other checks needed for validation
+        # Dependency analysis, following rules:
+        # No child has a write dependency to the loop variable.
+        # Find variable access info for the loop variable and step
+        refs = VariablesAccessInfo(node.children[0])
+        node_vars = VariablesAccessInfo()
+        if refs is not None:
+            node_vars.merge(refs)
+        refs = VariablesAccessInfo(node.children[1])
+        if refs is not None:
+            node_vars.merge(refs)
+        # The current implementation of BlockedLoopTrans does not allow
+        # the step size to be non-constant, so it is ignored.
+
+        # Add the access pattern to the node variable name
+        node_vars.add_access(Signature(node.variable.name),
+                             AccessType.WRITE, self)
+        node_vars.add_access(Signature(node.variable.name),
+                             AccessType.READ, self)
+
+        node_sigs = node_vars.all_signatures
+
+        # Find the Loop code's signatures
+        child_refs = VariablesAccessInfo(node.children[3])
+        child_sigs = child_refs.all_signatures
+
+        for ref1 in node_sigs:
+            if ref1 not in child_sigs:
+                continue
+            access2 = child_refs[ref1]
+
+            # If access2 is a write then we write to a loop variable
+            if access2.is_written():
+                raise TransformationError("Cannot apply a BlockedLoopTrans "
+                                          "to a loop where loop variables are "
```

Comment:
"to this loop because the boundary variable '{0}' is written..." and then format(access2.name)

Reply:
Done